app/src/main/java/com/mapbox/maps/testapp/examples/InsetMapActivity.kt

```diff
@@ -51,57 +51,61 @@ class InsetMapActivity : AppCompatActivity(), OnCameraChangeListener {
     if (insetMapFragment == null) {
       // Create fragment transaction for the inset fragment
       val transaction = supportFragmentManager.beginTransaction()
-
       insetMapFragment = MapFragment()
-      insetMapFragment.getMapAsync {
-        insetMapboxMap = it
-        insetMapboxMap.loadStyleUri(
-          styleUri = STYLE_URL
-        ) { style ->
-          val source = geoJsonSource(BOUNDS_LINE_LAYER_SOURCE_ID) {
-            feature(Feature.fromGeometry(LineString.fromLngLats(getRectanglePoints())))
-          }
-          style.addSource(source)
-
-          // The layer properties for our line. This is where we make the line dotted, set the color, etc.
-          val layer = lineLayer(BOUNDS_LINE_LAYER_LAYER_ID, BOUNDS_LINE_LAYER_SOURCE_ID) {
-            lineCap(LineCap.ROUND)
-            lineJoin(LineJoin.ROUND)
-            lineWidth(3.0)
-            lineColor(Color.YELLOW)
-            visibility(Visibility.VISIBLE)
-          }
-          style.addLayer(layer)
-          updateInsetMapLineLayerBounds(style)
-        }
-        insetMapFragment.getMapView()?.apply {
-          logo.enabled = false
-          scalebar.enabled = false
-          attribution.enabled = false
-          compass.enabled = false
-
-          gestures.updateSettings {
-            scrollEnabled = false
-            pinchToZoomEnabled = false
-          }
-        }
-      }
-
+      setInsetMapStyle(insetMapFragment)
       // Add fragmentMap fragment to parent container
       transaction.add(R.id.mini_map_fragment_container, insetMapFragment, INSET_FRAGMENT_TAG)
       transaction.commit()
+    } else {
+      setInsetMapStyle(insetMapFragment)
```

Comment:
we can simply move this line out of the if else logic? as both condition will execute the `setInsetMapStyle`

Reply:
yeah correct 😄